src/ui/osx/TogglDesktop/test2/UnsupportedNotice.m

```diff
@@ -0,0 +1,52 @@
+//
+//  UnsupportedNotice.m
+//  TogglDesktop
+//
+//  Created by Indrek Vändrik on 23/11/2018.
+//  Copyright © 2018 Alari. All rights reserved.
+//
+
+#import "UnsupportedNotice.h"
+
+@implementation UnsupportedNotice
+
++ (instancetype)sharedInstance
+{
+	static UnsupportedNotice *instance;
+	static dispatch_once_t onceToken;
+
+	dispatch_once(&onceToken, ^{
+					  instance = [[UnsupportedNotice alloc] init];
+				  });
+	return instance;
+}
+
+- (void)showNotice
+{
+	NSDate *today = [NSDate date];
+	NSDate *deadline = [NSDate dateWithString:@"2019-01-01 00:00:00 +0000"];
+	NSComparisonResult result = [today compare:deadline];
+
+	NSString *title = [NSString stringWithFormat:@"Mac OS X version not supported"];
+	NSString *text = @"";
+
+	if (result == NSOrderedAscending)
+	{
+		text = @"Toggl Desktop will stop working with your current version of OS X from the 1st of January 2019.\n\n";
+	}
+
+	text = [text stringByAppendingString:@"Please upgrade your system to macOS 10.11 or later to continue using Toggl Desktop."];
+
+	[self setMessageText:title];
+	[self setInformativeText:text];
+	[self addButtonWithTitle:@"OK"];
+	[self setAlertStyle:NSWarningAlertStyle];
+	[self runModal];
+
+	if (result != NSOrderedAscending)
```

Comment:
The code will terminal **before** the alert actually appear.
This one might be a mistake. According to our requirement
```
it closes the app immediately after the alert is closed.
```
I think that should be to make sure that user is able to read the reason why the app will terminal.
```objective-c
[alert beginSheetModalForWindow:self.window completionHandler:^(NSModalResponse returnCode) {
    if (returnCode == NSModalResponseOK) {
        // Handle terminal logic here
        return;
    }
   
}];
```

Reply:
I double checked and nothing is executed that is after the `[self runModal];` before the "OK" is pressed in the modal. This should not be an issue.